src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterObject.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.Json.Serialization.Policies;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal sealed class JsonValueConverterObject : JsonValueConverter<object>
+    {
+        public override bool TryRead(Type valueType, ref Utf8JsonReader reader, out object value)
+        {
+            if (JsonDocument.TryParseValue(ref reader, out JsonDocument document))
+            {
+                value = document.RootElement.Clone();
+                document.Dispose();
+                return true;
+            }
+            else
+            {
+                value = default;
+                return false;
+            }
+        }
+
+        public override void Write(object value, Utf8JsonWriter writer)
+        {
+            // Should only get here if the object is actually System.Object.
```

Comment:
Is it possible to do a debug.assert type check to validate the comment is true?

Reply:
I can, didn't think it was that useful as it shouldn't happen and we throw on the next line. If you have `new object()` assigned to an object field it might get here- I was just trying to say we can never go from `object` to `JsonElement` when serializing. I'm not even sure it's possible to get in here period. I'll write a test to see what happens in that case and I'll try to be more specific.

Reply:
Doesn't get in here even in that case. Changing to `InvalidOperationException`.